be/src/exec/pipeline/scan_operator.cpp

```diff
@@ -255,6 +259,39 @@ Status ScanOperator::_pickup_morsel(RuntimeState* state, int chunk_source_index)
     return Status::OK();
 }
 
+Status ScanOperator::_capture_tablet_rowsets() {
```

Comment:
put this function into ScanOperatorFactory and invoke it in ScanOperatorFactory::prepare, would be better?

Reply:
`ScanOperatorFactory` doesn't own morsels, and each `ScanOperator` instance has its own different morsels.